apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -1,10 +1,81 @@
-import React, { ReactElement } from "react";
+import React, { ReactElement, useEffect, useState } from "react";
+import Pill from "./Pill";
+import { Mode } from "../../__v3api";
+import { uniqueId } from "lodash";
+
+export const BUS = "bus";
+export const SUBWAY = "subway";
+export const FERRY = "ferry";
+export const COMMUTER_RAIL = "commuter_rail";
+
+// TODO replace with real data
+const departures: any[] = [
+  {
+    headsign: "Fields Corner",
+    routeNumber: 210,
+    mode: "bus"
+  },
+  {
+    headsign: "Columbian Square",
+    routeNumber: 226,
+    mode: "commuter_rail"
+  },
+  {
+    headsign: "Quincy Center",
+    routeNumber: 230,
+    mode: "subway"
+  },
+  {
+    headsign: "Montello",
+    routeNumber: 230,
+    mode: "ferry"
+  },
+  {
+    headsign: "South Shore Plaza",
+    routeNumber: 236,
+    mode: "bus"
+  }
+];
+
+const filterDeparturesByMode = (
+  departures: any[],
+  mode: Mode | "all"
+): any[] => {
+  if (mode === "all") {
+    return departures;
+  }
+  return departures.filter(d => d.mode === mode);
+};
 
 const StopPageRedesign = ({
   stopId
 }: {
   stopId: string;
 }): ReactElement<HTMLElement> => {
+  const [selectedMode, setSelectedMode] = useState<"all" | Mode>("all");
+  // TODO replace type with actual data type
+  const [filteredDepartures, setFilteredDepartures] = useState<any[]>([]);
+
+  const [hasBusRoutes, setHasBusRoutes] = useState<boolean>(false);
+  const [hasCommuterRailRoutes, setHasCommuterRailRoutes] = useState<boolean>(
+    false
+  );
+  const [hasSubwayRoutes, setHasSubwayRoutes] = useState<boolean>(false);
+  const [hasFerryRoutes, setHasFerryRoutes] = useState<boolean>(false);
+
+  useEffect(() => {
+    setHasBusRoutes(filterDeparturesByMode(departures, BUS).length > 0);
+    setHasCommuterRailRoutes(
+      filterDeparturesByMode(departures, COMMUTER_RAIL).length > 0
+    );
+    setHasSubwayRoutes(filterDeparturesByMode(departures, SUBWAY).length > 0);
+    setHasFerryRoutes(filterDeparturesByMode(departures, FERRY).length > 0);
+  }, [departures]);
```

Comment:
suggestion: Since the logic is the same for all modes, it would be more legible to consolidate the modes into one object and update that. e.g. something like (untested, sorry)

```tsx
const hasRouteTypes, setRouteTypes = useState<Record<keyof ModeEnum, boolean>>({}) // should pre-populate with false values probably

useEffect(() => {
  const routeTypes = {};
  // if you happen to have an Enum with each mode listed you can iterate it!
  for (let type in ModeEnum) {
    routeTypes[type] = (filterDeparturesByMode(departures, type).length > 0);
  }
  setRouteTypes(routeTypes);
}, [departures]);

// get the value via hasRouteTypes["SUBWAY"] or some such
```